src/Raven.Server/Documents/Sharding/ShardedDatabaseContext.cs

```diff
@@ -87,35 +93,50 @@ public ShardedDatabaseContext(ServerStore serverStore, DatabaseRecord record)
 
         public IDisposable AllocateOperationContext(out JsonOperationContext context) => ServerStore.ContextPool.AllocateOperationContext(out context);
 
-        public void UpdateDatabaseRecord(RawDatabaseRecord record, long index)
+        public void UpdateDatabaseRecord(DatabaseRecord record, long index)
+        {
+            try
+            {
+                DatabasesLandlord.NotifyFeaturesAboutStateChange(record, index, _orchestratorStateChange);
+                RachisLogIndexNotifications.NotifyListenersAbout(index, e: null);
+            }
+            catch (Exception e)
+            {
+                RachisLogIndexNotifications.NotifyListenersAbout(index, e);
+                throw;
+            }
+        }
+
+        private void OnDatabaseRecordChange(DatabaseRecord record, long index)
         {
             UpdateConfiguration(record.Settings);
 
             if (DictionaryExtensions.KeysEqual(record.Sharding.Shards, _record.Sharding.Shards) == false)
             {
-                using (var se = ShardExecutor)
-                {
-                    ShardExecutor = new ShardExecutor(ServerStore, record, record.DatabaseName);
-                }
+                ShardExecutor = new ShardExecutor(ServerStore, record, record.DatabaseName);
             }
 
             if (EnumerableExtension.ElementsEqual(record.Sharding.Orchestrator.Topology.Members, _record.Sharding.Orchestrator.Topology.Members) == false)
             {
-                using (var ne = AllOrchestratorNodesExecutor)
-                {
-                    AllOrchestratorNodesExecutor = new AllOrchestratorNodesExecutor(ServerStore, record);
-                }
+                AllOrchestratorNodesExecutor = new AllOrchestratorNodesExecutor(ServerStore, record);
             }
 
             Indexes.Update(record, index);
 
             SubscriptionsStorage.Update(record);
 
             Interlocked.Exchange(ref _record, record);
+        }
+
+        private void OnUrlChange(DatabaseRecord record, long index)
+        {
```

Comment:
Here needs a comment saying that we explicitly release the old versions for the GC to cleanup, since this is expected to be rare